turbopack/crates/turbopack-core/src/module_graph/module_batches.rs

```diff
@@ -513,28 +503,43 @@ pub async fn compute_module_batches(
 
         let mut edges_count = 0;
 
-        // Since batches can only have references followed by a list of parallel modules, we need to
-        // split batches that have modules before references.
+        // Since batches can only have references followed by a list of parallel chunkable modules,
+        // we need to split batches that have modules before references.
         for i in 0..pre_batches.batches.len() {
             let items = take(&mut pre_batches.batches[i].items);
             let mut new_items =
                 FxIndexSet::with_capacity_and_hasher(items.len(), Default::default());
             enum Mode {
-                ParallelModule,
+                ParallelChunkableModule,
                 Other,
             }
             let mut mode = Mode::Other;
             for item in items {
-                match (&mode, &item) {
-                    (_, PreBatchItem::ParallelModule(_)) => {
-                        mode = Mode::ParallelModule;
+                let chunkable_module = if let PreBatchItem::ParallelModule(module) = &item {
+                    ResolvedVc::try_downcast::<Box<dyn ChunkableModule>>(*module)
+                } else {
+                    None
+                };
+                let item = if let PreBatchItem::ParallelModule(module) = item {
+                    if chunkable_module.is_some() {
+                        PreBatchItem::ParallelModule(module)
+                    } else {
+                        pre_batches.single_module_entries.insert(module);
```

Comment:
That line is intentional? Non-chunkable modules just get their own batch (=no batch)?